doc_src/cmds/path.rst

```diff
@@ -0,0 +1,402 @@
+.. _cmd-path:
+
+path - manipulate and check paths
+=================================
+
+Synopsis
+--------
+
+::
+
+    path basename GENERAL_OPTIONS [PATH...]
+    path dirname GENERAL_OPTIONS  [PATH...]
+    path extension GENERAL_OPTIONS [PATH...]
+    path filter GENERAL_OPTIONS [(-v | --invert)] \
+        [-d] [-f] [-l] [-r] [-w] [-x] \
+        [(-t | --type) TYPE] [(-p | --perm) PERMISSION] [PATH...]
+    path is GENERAL_OPTIONS [(-v | --invert)] [(-t | --type) TYPE] \
+        [-d] [-f] [-l] [-r] [-w] [-x] \
+        [(-p | --perm) PERMISSION] [PATH...]
+    path normalize GENERAL_OPTIONS [PATH...]
+    path resolve GENERAL_OPTIONS [PATH...]
+    path change-extension GENERAL_OPTIONS EXTENSION [PATH...]
+    path sort GENERAL_OPTIONS [(-v | --invert)] \
+        [-u | --unique] [--key=basename|dirname|path] [([PATH...]
+
+    GENERAL_OPTIONS := [(-z | --null-in)] [(-Z | --null-out)] [(-q | --quiet)]
```

Comment:
Are we planning to add null switches to other commands as well? I don't see `path` and `string` being all that different in this regard.

Reply:
I believe we should add them to string as well, yes.

The main reason this is in path first is because it's *necessary* for path to be complete - NUL is the only byte you can use to separate paths unambiguously, and so it exists in a lot of commands (`find -print0`). Whereas string already passes through NULs from stdin, and has `string split0` that you can use to *work* with NUL-delimited strings by passing them as arguments.

Unfortunately `string` won't be able to use the PATH_MAX trick, because strings are arbitrary.